File: leetcode/python/removeDuplicatesSortedArray.py
# remove duplicates from sorted array

nums = [1, 3, 3, 5, 7, 8, 8, 19, 35, 57, 57, 67, 85]

def removeDuplicates(nums) -> int:
  nums[:] = sorted(set(nums))
  return len(nums)

print(removeDuplicates(nums))

File: leetcode/javascript/removeDuplicatesSortedArray.js
// remove duplicates from sorted array

let array = [0, 2, 6, 6, 10, 19, 44, 55, 55, 65, 67, 75, 88, 88, 132, 143, 143];

let removeDuplicatesSortedArray = (array) => {
	let first = 0;
	let second = 1;
	while (array[second] != "-") {
		if (array[first] === array[second]) {
			array.push(array.splice(second, 1).pop());
			array[array.length - 1] = "-";
		} else {
			first++;
			second++;
		}
	}
	return second;
};

console.log(removeDuplicatesSortedArray(array));
